src/sqlfluff/core/rules/base.py

```diff
@@ -433,6 +438,32 @@ def _process_lint_result(res):
 
     # HELPER METHODS --------
 
+    def is_final_segment(self, context: RuleContext) -> bool:
```

Comment:
Nice! I should look some time -- I think L003 may be able to use your function instead of the similar function [here](https://github.com/sqlfluff/sqlfluff/blob/253cfaaf058e364336234cc43f3775224de41dbd/src/sqlfluff/rules/L003.py#L235).

Reply:
awesome, looks like it should be a pretty straight swap! 😄